codice/ddf
# DDF-6215 Upgrade to Karaf 4.2.9

#### What does this PR do?
Upgrades to Karaf 4.2.9, and along with it upgrades a bunch of other dependencies to match the version that Karaf uses.

There was an attempt to upgrade previously, which ran into a number of issues (https://github.com/codice/ddf/issues/5798). I ran into a few of the same issues for this PR. Here's the fixes: 

> Some change, somewhere, has led to a problem with the bnd library we use (stacktrace here). There are two versions that I can see in the system and neither seems to correctly parse some classfile.

This was a problem parsing one of the cxf feature definitions. I copied the fix from https://github.com/apache/cxf/pull/602

> Some change has apparently altered startup order such that when running component tests, many (perhaps all) of our filters cannot be injected because their target servlet contexts have already started.

I changed our DelegateServletFilter to be a Jetty Handler, so it doesn't need to be injected into each individual servlet context. There was no way I could find to access the servlet contexts before they were started. The handler approach accomplishes the same thing - running filters for all requests - but filters must now implement the `org.codice.ddf.platform.filter.http.HttpFilter` interface instead of `javax.servlet.Filter` (Jetty Handlers accept HttpServletRequests instead of ServletRequests).

This PR depends on: https://github.com/codice/thirdparty/pull/97

#### Who is reviewing it? 

@blen-desta @bakejeyner @garrettfreibott @stustison 

#### Select relevant component teams: 

@codice/security 

#### How should this be tested?
Verify:
* Full build with itests passes
* Error pages are still registered. If you refresh the `platform-paxweb-jettyconfig` bundle, you should see a bunch of messages indicating that error pages are being injected as the webapps restart. Also, produce an error in the UI (e.g. 403 by hitting /admin as guest) and verify you see the expected DDF error page.
* HttpFilters are run on all requests and after the SecurityFilters (WebSsoFilter, LoginFilter, etc.). You can turn up the logs for `org.codice.ddf.pax.web.jetty` to TRACE and you should see a message `Delegating to 5 HttpFilters` when you make a request.
* The metrics filter still works. Hit `/metrics` and verify the `ddf_platform_*` metrics look realistic.
* HttpFilters are run in the correct order: SecurityJavaSubjectFilter first, DoPrivilegedFilter last, anything else in between

#### Any background context you want to provide?
This is part of an ongoing effort to reduce the CVE level in DDF

#### What are the relevant tickets?
Fixes: #6215

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
